provider/baked/src/binary_search.rs

```diff
@@ -117,9 +117,20 @@ pub(crate) fn bake(
 pub struct Data<K: BinarySearchKey, M: DataMarker>(pub &'static [(K::Type, &'static M::Yokeable)]);
 
 impl<K: BinarySearchKey, M: DataMarker> super::DataStore<M> for Data<K, M> {
-    fn get(&self, id: DataIdentifierBorrowed) -> Option<&'static M::Yokeable> {
+    fn get(
+        &self,
+        id: DataIdentifierBorrowed,
+        attributes_prefix_match: bool,
+    ) -> Option<&'static M::Yokeable> {
         self.0
             .binary_search_by(|&(k, _)| K::cmp(k, id))
+            .or_else(|e| {
+                if attributes_prefix_match {
+                    Ok(e)
+                } else {
+                    Err(e)
+                }
+            })
```

Comment:
Question/Issue: Is this code path being tested? I think you need to check if the prefix matches, still; `binary_search_by` will return you an element that doesn't necessarily share the prefix

Reply:
It will be tested through `test-baked-source` if it's used, but I guess I should remove it? Or do we want to keep this as an option going forward?

Reply:
`test-baked-source` should test both the binary search path and the zerotrie path if we claim to support either one.

Reply:
We don't claim to support either. Currently binary search is dead code, I might add this as an option, I just don't want to delete it yet.